Source/Core/InputCommon/ControllerInterface/Win32/Win32.cpp

```diff
@@ -19,20 +19,27 @@
 
 constexpr UINT WM_DOLPHIN_STOP = WM_USER;
 
-static Common::Event s_done_populating;
-static std::atomic<HWND> s_hwnd;
-static HWND s_message_window;
+static Common::Event s_received_device_change_event;
+static HWND s_hwnd;            // Dolphin's render window
+static HWND s_message_window;  // Windows messaging window (hidden)
 static std::thread s_thread;
+static std::atomic<bool> s_first_pupulate_devices_asked;
 
 static LRESULT CALLBACK WindowProc(HWND hwnd, UINT message, WPARAM wparam, LPARAM lparam)
 {
   if (message == WM_INPUT_DEVICE_CHANGE)
   {
-    g_controller_interface.PlatformPopulateDevices([] {
-      ciface::DInput::PopulateDevices(s_hwnd);
-      ciface::XInput::PopulateDevices();
-    });
-    s_done_populating.Set();
+    // Windows automatically sends this message before we ask for it and "ready" to listen for it.
+    if (s_first_pupulate_devices_asked)
+    {
+      s_received_device_change_event.Set();
+      // TODO: we could easily use the message passed alongside this event, which tells
+      // whether a device was added or revoved, to avoid removing old, still connected, devices
```

Comment:
"still connected, devices" -> "still connected devices"

Reply:
Subjective, I think it's correct the way it is. Not a big deal I hope anyway...